dimod/constrained.py

```diff
@@ -773,9 +774,112 @@ def vartype(self, v: Variable) -> Vartype:
         """The vartype of the given variable."""
         return self.variables.vartype(v)
 
+    @classmethod
+    def from_lp_file(cls,
+                     fp: Union[BinaryIO, ByteString],
+                     lower_bound_default: Optional[int] = None,
+                     upper_bound_default: Optional[int] = None) -> "ConstrainedQuadraticModel":
+        """
+        Create a CQM model from a LP file.
+        Args:
+            fp: file-like object in the LP format
+            lower_bound_default: in case lower bounds for integer are not given, this will be the default
+            upper_bound_default: in case upper bounds for integer are not given, this will be the default
 
-CQM = ConstrainedQuadraticModel
+        Returns:
+            The model encoded in the LP file, converted to ConstrainedQuadraticModel dimod object
+        """
+
+        grammar = make_lp_grammar()
+        parse_output = grammar.parseFile(fp)
+        variables_info = get_variables(parse_output, lower_bound_default, upper_bound_default)
```

Comment:
It looks like with relatively few changes you could make `get_variables` an iterator rather than returning a dict. Then do something like
```
for v, vartype, lb, ub in get_variables(parse_output):
    if vartype is Vartype.INTEGER:
        obj.add_variable(vartype, v, lb, ub)
    elif vartype is Vartype.BINARY:
        obj.add_variable(vartype, v)
    else:
        RuntimeError("unexpected vartype")  # or maybe ValueError if this hasn't been checked before
```

Since `obj` will then have all of the variables, you can use it's `.variables` `.lower_bound()` and `.upper_bound` as the source of truth in later steps.